core/src/test/java/org/apache/brooklyn/core/catalog/internal/SpecParameterInMetaTest.java

```diff
@@ -43,18 +49,40 @@
 public class SpecParameterInMetaTest {
     private ManagementContext mgmt;
     private BrooklynCatalog catalog;
-    private String spec;
+    private String specId;
 
     @BeforeMethod(alwaysRun=true)
     public void setUp() {
         mgmt = LocalManagementContextForTests.newInstanceWithOsgi();
         catalog = mgmt.getCatalog();
-        spec = TestToSpecTransformer.registerSpec(EntitySpec.create(BasicEntity.class));
+        StaticTypePlanTransformer.forceInstall();
+        PlanToSpecFactory.forceAvailable(TestToSpecTransformer.class, JavaCatalogToSpecTransformer.class);
+        specId = StaticTypePlanTransformer.registerSpec(EntitySpec.create(BasicEntity.class));
     }
 
+    @AfterMethod(alwaysRun=true)
+    public void tearDown() {
+        StaticTypePlanTransformer.clearForced();
+        PlanToSpecFactory.clearForced();
+    }
+
+    @Test
+    public void testCanRetrieveWithNew() {
+        AbstractBrooklynObjectSpec<?, ?> spec = mgmt.getTypeRegistry().createSpecFromPlan(null, specId, null, null);
+        Assert.assertNotNull(spec);
+    }
+
+    // it's not actually added to the catalog; probably it would be cleaner if it is;
```

Comment:
The idea of `TestToSpecTransformer` is to convert a plan to a pre-registered spec. It's not working on types but on plans. This means that it's able to convert a plan it understands (be it coming from a catalog item or used as an app plan) to a spec. The plan itself may be added as a catalog item, but it's a separate matter.

A plan refers to a type, but is not itself a type, it's a composition of types. Calling `registerSpec` tells the transformer how to interpret the plan, not what the spec for a specific type is.